mlcustomize/inject_virtio_win.ml

```diff
@@ -487,43 +486,40 @@ and virtio_iso_path_matches_guest_os t path =
       else if pathelem "amd64" then "x86_64"
       else raise Not_found in
 
-    let is_client os_variant = os_variant = "Client"
-    and not_client os_variant = os_variant <> "Client"
-    and any_variant os_variant = true
-    and any_osinfo osinfo = true in
-    let p_os_major, p_os_minor, match_os_variant, match_osinfo =
+    let match_osinfo =
       if pathelem "xp" then
-        (5, 1, any_variant, any_osinfo)
+        ((=) "winxp")
       else if pathelem "2k3" then
-        (5, 2, any_variant, any_osinfo)
+        (* matches win2k3 and win2k3r2 *)
+        (String.is_prefix "win2k3r2")
```

Comment:
I think it's best to just explicitly match what you want.  Replace both lines with just:
```
(function "win2k3" | "win2k3r2" -> true | _ -> false)
```